main.go

```diff
@@ -40,14 +40,20 @@ import (
 )
 
 const (
-	namespaceEnvVar = "NAMESPACE"
+	namespaceEnvVar    = "NAMESPACE"
+	configSecretEnvVar = "CONFIG"
```

Comment:
I think the naming here is problematic. 

First of all the vars are not env vars but env vars name so a better prefix should be `EnvVarName` (I did not understand it was the name until I read the rest of the code which is a bad sign for readability)

Second, the "CONFIG" env var is incorrectly named, this is not the name of the env var that holds some general config, or the only config details for that matter. It is the env var that specifies the name of the AddOnParamsSecret and should be named as such, for example 
```addonParamsSecretEnvVarName = "ADDON_PARAMS_SECRET"```

But I have another more "radical" suggestion: Why not just hold an env var for the add-on name, like
```addonNameEnvVarName="ADDON_NAME"```
Now we can use it to build the secret name below and moving forward also use it for other tasks that need the addon name (For example reading the config-map that is used to indicate uninstallation)